x-pack/plugin/rest-compatibility/src/main/java/org/elasticsearch/compat/CompatibleVersionPlugin.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.compat;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.RestCompatibilityPlugin;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CompatibleVersionPlugin extends Plugin implements RestCompatibilityPlugin {
+
+    private static final Pattern COMPATIBLE_API_HEADER_PATTERN = Pattern.compile(
```

Comment:
this will change to using XContentType.parseVersion once https://github.com/elastic/elasticsearch/pull/61427 is merged